src/net.h

```diff
@@ -991,9 +991,12 @@ class NetEventsInterface
     /** Mutex for anything that is only accessed via the msg processing thread */
     static Mutex g_msgproc_mutex;
 
-    /** Initialize a peer (setup state, queue any initial messages) */
+    /** Initialize a peer (setup state) */
     virtual void InitializeNode(CNode& node, ServiceFlags our_services) = 0;
 
+    /** Queue initial messages for peer (only relevant for outbound connections) */
+    virtual void SendInitialMessages(CNode& node) = 0;
```

Comment:
I think I'd be nicer to not change this interface and I believe it would be trivial to just use `SendMessages`? i.e. push the version message to outbound peers at the start of `SendMessages` (if it hasn't been pushed already).

Reply:
Please put that state on `Peer` if you go this route.